api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package envoy.extensions.transport_sockets.http_11_proxy.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.transport_sockets.http_11_proxy.v3";
+option java_outer_classname = "UpstreamHttp11ConnectProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/http_11_proxy/v3;http_11_proxyv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Upstream HTTP/1.1 Proxy]
+// [#extension: envoy.transport_sockets.http_11_proxy]
+
+// Configuration for HTTP/1.1 proxy transport sockets.
+// If this is configured and an intermediate filter adds proxy metadata to the
+// stream info then
+//   - Upstream connections will be directed to the specified proxy address rather
+//     than the host's address
```

Comment:
Does this avoid the DNS lookup for the server's address? I'm guessing not. I think that would be a good optimization to add because the DNS lookup might fail (in the worst case) or might take unnecessary time (in the best case). Not something to do now, but something to maybe add a TODO / issue about?

Reply:
That's a great call out.  It does not - I think we'll have to do that in the DFP filter since the CONNECT code doesn't interact with DNS.  I've added a TODO to do this as a follow-up.